### PR TITLE
Attach terminal for shipyard.

### PR DESCRIPTION
**Pull request by ukd1**
@diffoperator and I hacked this up...@smathieu did a ton of work on https://github.com/rainforestapp/Docker-Terminal. Enjoy!

It requires https://github.com/dotcloud/docker/pull/1146 to be merged before it will be of general use!

### CODE REVIEW

**Comment by ehazlett** [comment]
Hey awesome work!  Thanks!  Which version of docker are you using?  If I pull master, merge the 1146 branch, I only get "session terminated" when trying to attach?

File: static/js/docker_terminal.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+window.docker = (function(docker) {
+  docker.terminal = {
+    startTerminalForContainer: function(container) {
+      var term = new Terminal();
+      term.open();
+
+      var wsUri = "ws://storagelols:4243/v1.3/containers/" + 
+        container + 
+        "/attach/ws?logs=1&stderr=1&stdout=1&stream=1&stdin=1";
+
```
Comment (sarda-nikhil) [inline static/js/docker_terminal.js:10]: @ukd1 This should probably not be hardcoded.

**Comment by sarda-nikhil** [comment]
We were using the v1.3 api. I think that the docker_terminal.js should be moved from static and embedded into the template and the uri should be passed through as a template token.

**Comment by ehazlett** [comment]
Yeah I'm going to clean that up.

**Comment by ehazlett** [comment]
Hey I wanted to keep this out of master since Docker hasn't merged the websocket functionality yet.  I'm going to close this one and use https://github.com/ehazlett/shipyard/pull/9 as the PR for this.

I got it working and it's crazy awesome!  Thanks again!

File: static/js/docker_terminal.js (hunk 1 above)
Comment (ukd1) [inline static/js/docker_terminal.js:10]: :+1: hehe

**Pull request closed without merging**